tests/main/test_mp.py

```diff
@@ -115,38 +114,92 @@ def test_mp(tmp_path: Path):
 
     with tempfile.TemporaryDirectory() as fp:
         config.experiment_dir = fp
-        assert_error_msg(
-            lambda: ParallelTrainer(wrapper=wrapper, run_config=config),
-            "Device must be set to 'cuda' if `gpu_mb_per_experiment` > 0",
-        )
         config.device = "cuda"
-        config.gpu_mb_per_experiment = 1e12
-        config.cpus_per_experiment = 1e12
-        out, err = capture_output(
-            lambda: ParallelTrainer(wrapper=wrapper, run_config=config)
-        )
-        assert (
-            "Expected GPU memory utilization" in out
-            and "Expected CPU core util. exceed system capacity" in out
-        )
-
+        if torch.cuda.is_available():
+            config.gpu_mb_per_experiment = 1e12
+            config.cpus_per_experiment = 1e12
+            out, err = capture_output(
+                lambda: ParallelTrainer(wrapper=wrapper, run_config=config)
+            )
+            assert (
+                    "Expected GPU memory utilization" in out
+                    and "Expected CPU core util. exceed system capacity" in out
+            )
+        else:
+            assert_error_msg(
+                lambda: ParallelTrainer(wrapper=wrapper, run_config=config),
+                "Could not find a torch.cuda installation on your system."
+
+            )
     config.experiment_dir = tmp_path
-    config.device = "cuda"
+    if not torch.cuda.is_available():
+        config.device = "cpu"
     config.gpu_mb_per_experiment = 0.001
     config.cpus_per_experiment = 0.001
     ablator = ParallelTrainer(wrapper=wrapper, run_config=config)
-    ablator.gpu = 1 / config.concurrent_trials
+    if torch.cuda.is_available():
+        ablator.gpu = 1 / config.concurrent_trials
```

Comment:
what is the reason for this assignment?

Reply:
Without checking for available `cuda`, ray will raise this error: `Error: No available node types can fulfill resource request {'GPU': 0.1}. Add suitable node types to this cluster to resolve this issue.` and the test will be stuck

Reply:
In case later we can run these tests on an environment that has cuda support

Reply:
Why not have it set through `config.gpu_mb_per_experiment` ?